content/general/limits.textile

```diff
@@ -0,0 +1,306 @@
+
+- Local limits := apply to a local part of the platform, such as an individual channel or connection. Examples of local limits include the maximum number of channels per connection and the maximum publish rate on a channel. Email notifications are not sent for local limits.
+- Global limits := apply to limits that effect your whole account, such as peak connections. Global limits are an aggregate of usage across all applications in your account. Email notification warnings will be sent when a global limit is close to being reached, or has been hit.
+
+h3(#instantaneous-rates). Instantaneous rate limits
+
+Instantaneous rate limits are based on the rate of an operation per second. If the limit is exceeded then the operation will be suppressed. 
+
+Rate suppression is calculated on a rolling probabilistic basis. For example, the limit on a self-service package for publishing messages into a queue is 100 messages per second. If a queue rule is attempting to publish 200 message per second into the queue, each message will have a 50% chance of being rejected. The suppression probability is continuously updated based on the publishing rate.
+
+Instantaneous rate limits only have a hard limit. Once the hard limit rate has been exceeded then message suppression will occur. As soon as the rate drops below the hard limit threshold, the suppression probability will decrease to zero.
+
+h3(#restrictions). Account restrictions
+
+Account restrictions prevent the creation of additional resources for the limit that was hit. All other operations can continue.
+
+For example, if the peak connection hard limit is hit then no more connections can be made to Ably until some of the existing ones disconnect and the total is back below the hard limit. If an hourly hard limit is hit, such as the number of API requests, then no more API requests can be made until the following hour. 
+
+Limits that can result in an account restriction have a soft limit and a hard limit. "Quota limits":#quota use the pre-paid quota as the soft limit. A restriction doesn't come into effect until the hard limit is hit. Usage above the soft limit for self-service and business account quota limits are charged as overages. Limits unrelated to quota limits, such as token requests, can still experience an account restriction, however any usage over the soft limit is not chargeable. 
+
+Account restrictions are lifted automatically once the resource is brought below the hard limit. For connections this means disconnecting unnecessary connections and for channels, detaching from any channels not in use. Hourly-based account restrictions such as API requests are automatically lifted the following hour. 
+
+h3(#blocks). Account block
```

Comment:
From the user's perspective, is there a particular reason to distinguish between 'Account Restrictions' and 'Account Block'? From what I can tell both are effectively hard limits on a parameter of Ably which stops it being used until either usage is reduced (channel limit) or the time period of the limit has ended (API requests limit/hourly message limit).

Personally I'd view from a user's perspective the more important differentiators are:
- Time-based limits (second/hour/month), and when do each of these reset (and highlighting that they do reset)
- - Instantaneous global limits then would be somewhat a subsection of this, mentioning the probabilistic rejections
- Quantity caps on resources (channels/connections/number of queues/anything else we have which has a max number where the value can go +-)

I've not been part of conversations on this topic though so apologies if I've missed any important reasoning for the current structure.

Reply:
Yes, I was thinking the same and whether we should just describe the limits as volumetric limits (messages, channels, connections) which result in restrictions being applied; and rate limits (both local and global) which result in rate limit enforcement by rejection of individual requests.

Reply:
This is an interesting point, and something I did consider when writing this. Let me re-write those sections whilst I'm separating out local/global rate limits and see how it reads.

Reply:
I've split out the section on instantaneous rate limits (and renamed it to just `rate limits`) and created sections for time and quantity based limits instead. I kept rate limits separate to time because they are enforced differently.

I think this is much clearer but let me know your thoughts.